src/lib/storage/boot_requirements_strategies/base.rb

```diff
@@ -31,9 +38,14 @@ module BootRequirementsStrategies
       class Base
         include Yast::Logger
 
+        using Refinements::Disk
+        using Refinements::DevicegraphLists
+
         def initialize(settings, disk_analyzer)
           @settings = settings
           @disk_analyzer = disk_analyzer
+          @root_disk = @disk_analyzer.devicegraph.disks.with(name: settings.root_device).first
```

Comment:
I also think this is clumsy. IMO it would help if we'd pass more the disk objects around instead of just device names. Then you don't constantly have to look up the actual object in some strange collection thing.